fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgresResourceReferenceDAO.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.postgresql;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.persistence.jdbc.dao.api.ICommonTokenValuesCache;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceReferenceDAO;
+import com.ibm.fhir.persistence.jdbc.dto.CommonTokenValue;
+
+
+/**
+ * Postgres-specific extension of the {@link ResourceReferenceDAO} to work around
+ * some SQL syntax and Postgres concurrency issues
+ */
+public class PostgresResourceReferenceDAO extends ResourceReferenceDAO {
```

Comment:
Note, we are inconsistent on the prefix for these classes.  I much prefer Postgres vs PostgreSQL

Also, this is a migration class?

Reply:
Yeah Postgres is much easier to read. The `postgresql` package name is legacy. What do you mean migration class? It's core to 1366 persistence.

Reply:
Raised issue #1636.

Reply:
sounds good - I think the way I tried to read this it came out of order.